stdlib/public/core/Slice.swift

```diff
@@ -229,6 +229,23 @@ extension Slice: Collection {
   }
 }
 
+extension Slice {
+  @inlinable
```

Comment:
Ditto here -- `@inlinable` introduces a new ABI symbol so it would need to come with availability. Aeic allows us to add the new entry point without having to also declare availability for it.

(The new definition will still have an effect on the ABI through witness tables and the like, but the stdlib won't export a new symbol for it.)